### PR TITLE
modulegroups : contextmenus refinments

### PR DESCRIPTION
**Pull request by AlicVB**
As discussed in #8075 
this change : 
- put the list of all widgets/modules in a submenu to reduce contextmenu length
- add some padding on top of contextmenu titles as suggested by @Nilvus 

... but @Nilvus, I fear that I'll need your help in css for second level menu in basic widgets : it has a strange arrow and submenu is black...

### CODE REVIEW

**Comment by elstoc** [comment]
I would prefer if the menu came straight off the header rather than as a single sub-item - i.e. it should be "add widget (all) > sub-menu" rather than "add widget (other) > all available widgets > sub-menu". Similarly for the modules. 

I suspect the top and bottom arrows are standard gtk things for when the menu goes off a page. Looking at the documentation I can't see a way to hide them but you can alter their style by referencing `arrow.bottom` and `arrow.top`. As for the black text I'm not sure how to view GTK Inspector while these menus are open - I can't seem to alt-tab to it without closing the menu down.

**Comment by AlicVB** [comment]
> I would prefer if the menu came straight off the header rather than as a single sub-item - i.e. it should be "add widget (all) > sub-menu" rather than "add widget (other) > all available widgets > sub-menu". Similarly for the modules.

I've tried, but either we design it like the other sections "title" and that's strange, either we design it like a classic item, and we loose the differentiation between recommended and others... So it's the best compromise I've found, but I'm all open to other ideas...



> I suspect the top and bottom arrows are standard gtk things for when the menu goes off the screen

No, I speak about the arrow in end of lines of items with submenus. Here, it looks like this : 
capture_tmp

**Comment by Nilvus** [comment]
I will see all that this week-end (maybe tomorrow) but 2 things:

I agree with @elstoc about sub-menu and I don't understand what you find strange to set sub-menu directly in add widget (other)? Could you precise in a comment the code to change for me, to test that? I have better ideas when I test and change some things. Maybe I could propose something.

About glitches in sub-menus, I remember to deal with tricky things on sub-menus in quick styles access in darkroom. Don't loose time on that, I will check and try to find how to deal with that with CSS.

**Comment by elstoc** [comment]
> either we design it like a classic item, and we loose the differentiation between recommended and others...

I agree that once you start introducing sub-menus there's probably no nice neat option and tbh I'm probably still more in favour of the single long list.

How about if we just have a single "add" section and then the last item is just "show all". For example

```
delete widget
   list of widgets
   :
add widget
   list of recommended widgets (if present)
   :
   show all > sub-menu
```

You'll still get a bit of oddness when there aren't any recommended widgets but maybe we just live with that. Apart from abandoning the sub-menus I'm not sure I have a better idea.

> No, I speak about the arrow in end of lines of items with submenus

I didn't even spot that. I'm clearly not very observant. We'll leave that to the expert :)

**Comment by Nilvus** [comment]
> How about if we just have a single "add" section and then the last item is just "show all". For example
> 
> ```
> delete widget
>    list of widgets
>    :
> add widget
>    list of recommended widgets (if present)
>    :
>    show all > sub-menu
> ```
> 
> You'll still get a bit of oddness when there aren't any recommended widgets but maybe we just live with that. Apart from abandoning the sub-menus I'm not sure I have a better idea.

It's probably the simplest and best idea. I'm on for that proposal. We just need a little extra space between recommended list and show all submenu, like for title, so a padding top of 0.4em for show all line.

**Comment by Nilvus** [comment]
Just a thought: 'show other' should be better than 'show all' as all could be understood as other AND recommended.
'show other' is so other than recommended.

**Comment by elstoc** [comment]
I thought that @AlicVB was going to change (had changed?) the other menu so it really did show all (including recommended)

**Comment by AlicVB** [comment]
thanks for your comments...
last commit implement what you propose and fix the alphabetic order.
I've kept the names "all available modules/widgets" for submenu text, as I find "show all" not reflecting the action, but I may very well be wrong and so don't hesitate to insist for other strings ;)

@Nilvus : submenus entries can be tweaked with` #modulegroups-popup-item2` in css, and I've at least removed the unneeded left padding, but main issues remain ;)

**Comment by elstoc** [comment]
Apart from the known issues which I've no idea about, it's looking good now. The only thing I'd say is that I'm not convinced by the extra space above the sub-menu at the bottom. Perhaps we need another way to separate it from the other entries. Bold text maybe?

**Comment by AlicVB** [comment]
and here comes the power of css, as that space , the text styling, etc... are controlled by css under the name `#modulegroups-popup-item-all` :)
So I've no doubt that @Nilvus (or you) will finetune all this if there's a better solution !

**Review by elstoc — APPROVED** [review]

**Review by Nilvus — APPROVED** [review]

**Comment by Nilvus** [comment]
@AlicVB and @elstoc: I suggest to use italic for all available modules (hope it's good for you). I also success to deal with remaining glitches in submenus. I suggest to post a PR after that one is merged to fix all that. See the result:



For me, this is ready to be merged.

**Comment by TurboGit** [comment]
@Nilvus : I'll merge later today so you have time to work on the CSS part as the status of this PR makes it hard to use the module groups.

**Comment by Nilvus** [comment]
@TurboGit: The work is already done. I'm just waiting to see that PR merged before posting my PR. I want to avoid a rebase.

**Comment by TurboGit** [comment]
@Nilvus : Now merged  !

**Pull request merged**
